tests/helpers/test_check_config.py

```diff
@@ -42,19 +43,41 @@ def log_ha_config(conf):
         _LOGGER.debug("error[%s] = %s", cnt, err)
 
 
+def _assert_warnings_errors(
+    res: HomeAssistantConfig,
+    errors: list[CheckConfigError],
+    warnings: list[CheckConfigError],
+) -> None:
+    assert len(res.errors) == len(errors)
+    assert len(res.warnings) == len(warnings)
+
+    error_str = ""
+    warning_str = ""
+
+    for idx, error in enumerate(res.errors):
+        assert error == errors[idx]
+        error_str += error.message
```

Comment:
It's a bit weird that we build the expected string from the object that we want to test. If we just want to assert the relative positions of the string parts I guess it's ok.

Reply:
Yes, it's weird. It's to make it possible to use the `ANY` test helper, but I'll rewrite the affected test to not use the helper 👍